### PR TITLE
Added network metrics to list currently connected peers

### PR DESCRIPTION
**Pull request by dankostiuk**
# Description

Followed the prometheus metrics convention discussed in https://github.com/0xPolygon/polygon-sdk/pull/216 to add a new `metrics.go` to track network-related metrics. I added the ability to track currently-connected peers by setting the gauge value each time we hit addPeer or delPeer.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

**Review by zivkovicmilos — CHANGES_REQUESTED** [review]
Looks great 💯 

Please resolve the test panics (you probably haven't initialized the metrics struct somewhere in the tests) 👍

**Comment by zivkovicmilos** [comment]
@munna0908 

It would be great if we could update the prometheus metrics on the docs repo to include this change as well

**Review by zivkovicmilos — APPROVED** [review]
Looks great 💯 

Thank you for resolving the failing tests, and for the contribution of course 🙏

File: network/server.go [hunk 1]
```diff
@@ -370,6 +374,7 @@ func (s *Server) addPeer(id peer.ID) {
 	s.peers[id] = p
 
 	s.emitEvent(id, PeerConnected)
+	s.metrics.Peers.Set(float64(len(s.peers)))
```
Comment (munna0908) [inline network/server.go:396]: @dankostiuk I would suggest to acquire peersLock before reading the length. @zivkovicmilos Whats your take ?

**Review by munna0908 — CHANGES_REQUESTED** [review]

File: network/server.go (hunk 1 above)
Comment (zivkovicmilos) [reply network/server.go:396]: You're right 💯 

@dankostiuk please call s.numPeers() since `s.peers` requires concurrent access.

Comment (dankostiuk) [reply network/server.go:396]: Just made that change but it looks like addPeer/delPeer were already acquiring `s.peersLock.Lock()` - unless I am missing something..

Comment (munna0908) [reply network/server.go:396]: @dankostiuk  Yeah you're right, I missed it. Please revert back

**Review by munna0908 — APPROVED** [review]
LGTM 🚀

**Review by Kourin1996 — APPROVED** [review]
Looks good. Thank you for the PR 🙏

**Review by lazartravica — APPROVED** [review]

**Comment by zivkovicmilos** [comment]
Hey @dankostiuk,

Please resolve the merge conflicts and pull the latest changes from develop, so we can merge out this PR 🚀

**Pull request merged**